{
  isRunning=false;
  collector.close();
  try {
    FunctionUtils.closeFunction(userFunction);
  }
 catch (  Exception e) {
    throw new RuntimeException("Error when closing the function",e);
  }
}
